test/e2e/scenario/recorder/viewports.scenario.ts

```diff
@@ -0,0 +1,371 @@
+      })
+
+    /**
+     * window.ScrollX/Y on some devices/browsers are changed by pinch zoom
+     * We need to ensure that our measurements are not affected by pinch zoom
+     */
+    createTest('getScrollX/Y should not be affected by pinch scroll')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        const VISUAL_SCROLL_DOWN_PX = 60
+        const LAYOUT_SCROLL_AMOUNT = 20
+
+        await buildScrollablePage()
+        await performSignificantZoom()
+        await resetLayoutScroll()
+
+        const initialVisualViewport = await getVisualViewport()
+        const { scrollX: initialScrollX, scrollY: initialScrollY } = await getWindowScroll()
+
+        // Add Visual Viewport Scroll
+        await pinchScrollVerticallyDown(VISUAL_SCROLL_DOWN_PX)
+
+        // Add Layout Viewport Scroll
+        await layoutScrollTo(LAYOUT_SCROLL_AMOUNT, LAYOUT_SCROLL_AMOUNT)
+
+        const nextVisualViewport = await getVisualViewport()
+        const { scrollX: nextScrollX, scrollY: nextScrollY } = await getWindowScroll()
+
+        await browserExecute(() => {
+          document.dispatchEvent(new Event('scroll'))
+        })
+
+        const lastScrollData = (
+          await getLastRecord(events, (segment) => findAllIncrementalSnapshots(segment, IncrementalSource.Scroll))
+        ).data as ScrollData
+
+        // Height changes because URL address bar changes due to scrolling
+        const navBarHeightChange = nextVisualViewport.height - initialVisualViewport.height
+        expect(navBarHeightChange).toBeLessThanOrEqual(30)
+
+        // Visual Viewport Scroll should change without visual viewport affect
+        expectToBeNearby(lastScrollData.x, initialScrollX + LAYOUT_SCROLL_AMOUNT)
+        expectToBeNearby(lastScrollData.y, initialScrollY + LAYOUT_SCROLL_AMOUNT)
+        expectToBeNearby(lastScrollData.x, nextScrollX)
+        expectToBeNearby(lastScrollData.y, nextScrollY)
+
+        // Test the Tests - Check that the math works out
+        expect(initialScrollX).toBe(0)
+        expect(initialScrollY).toBe(0)
+
+        // ScrollX/Y must be defined as (page - offset)
+        expectToBeNearby(lastScrollData.x, nextVisualViewport.pageLeft - nextVisualViewport.offsetLeft)
+        expectToBeNearby(lastScrollData.y, nextVisualViewport.pageTop - nextVisualViewport.offsetTop)
+
+        // Both Layout + Visual scroll should affect "page" dimensions
+        expectToBeNearby(
+          nextVisualViewport.pageTop,
+          LAYOUT_SCROLL_AMOUNT + VISUAL_SCROLL_DOWN_PX - navBarHeightChange + initialVisualViewport.pageTop
+        )
+        expectToBeNearby(nextVisualViewport.pageLeft, LAYOUT_SCROLL_AMOUNT + initialVisualViewport.pageLeft)
+      })
+  })
+
+  describe('visual viewport properties', () => {
```

Comment:
IIRC, from last talk we said that we'll remove this test.
anything change?

Reply:
Resolved IRL